pr.test.js

```diff
@@ -14,6 +14,42 @@ test('returns null if no PR number is found in a commit message', () => {
     expect(extractPRNumber('Merge branch master into some/mockBranch')).toEqual(null)
 })
 
+test('searchPRByCommit returns a pr', async () => {
+    let sha = "123456789"
+    const config = { octokit: { search: {
+         issuesAndPullRequests: async (options) => {
+             return { data: {
+                        total_count: 1,
+                        items: [{number: 15, id: sha }],
+                        query: options.q
+                    }}
+         }
+    }}}
+    expect(searchPRByCommit(sha, config)).resolves.toEqual({number: 15, id: sha})
```

Comment:
It would be good to add a test case here that attempts to find a PR by a different SHA than the one supported by the test. This covers the case where `total_count` is non-zero but the specific SHA is different.

Reply:
As the code is now I'm relying on the query from github which is filtering on the commit-sha.  The assumption is this query will work and we would never have any results back that would not have that sha.  This test wouldn't work as we are not checking that the results actually have the correct sha.
I can add a verification to the code to ensure that the PR in the results does have the correct sha and then add this test but the assumption is that the github query would be doing it correctly.